app/src/main/kotlin/com/github/gotify/settings/SettingsActivity.kt

```diff
@@ -121,13 +128,15 @@ internal class SettingsActivity : AppCompatActivity(), OnSharedPreferenceChangeL
         }
 
         private fun restartApp() {
+            requireContext().getSystemService<ActivityManager>()?.appTasks?.forEach { task ->
+                task.finishAndRemoveTask()
+            }
             val packageManager = requireContext().packageManager
             val packageName = requireContext().packageName
             val intent = packageManager.getLaunchIntentForPackage(packageName)
             val componentName = intent!!.component
             val mainIntent = Intent.makeRestartActivityTask(componentName)
             startActivity(mainIntent)
-            Runtime.getRuntime().exit(0)
         }
```

Comment:
What does this change do?, why do we have to remove the appTasks here?

Reply:
To be honest, I think it is the simplest and most efficient way to directly modify the visibility with ActivityManager at the beginning.

Reply:
You mean like it was done in your first PR? I think I agree, now this feature could affect other parts of Gotify and your first change only affected the app, if the feature flag was enabled. Do you still have the changes from your first PR? The old PR https://github.com/gotify/android/pull/281 seems broken and doesn't have changes anymore.

Reply:
Yes, I have forced push the change in #281 to this PR.

Reply:
Looks good, I changed the signature of the method, to not be an extension method. And the implementation can use getOrNull, to remove the need for the takeIf.

Reply:
Thanks for your work! ❤️